src/ansible_navigator/configuration_subsystem/definitions.py

```diff
@@ -134,16 +138,21 @@ class SubCommand(SimpleNamespace):
     epilog: Union[None, str] = None
 
 
-class ApplicationConfiguration(SimpleNamespace):
+@dataclass
```

Comment:
It's usually preferable to have immutable data structures
```suggestion
@dataclass(frozen=True, slots=True)
```

Reply:
The settings are reevaluated and updated with every `:command`. If you get time, pull down the PR and let me know how it works with it frozen...  I'm curious.

Reply:
Got it. Is it the same for every class in this module? If yes, maybe it's worth explaining in a code comment at the very top.

Reply:
(feel free to "Resolve" this conversation when you read it)